### PR TITLE
Add workflow figures and technique descriptions

### LINKED ISSUE #944: Add workflow figures to document techniques
It would be nice to add workflow figures to document techniques, taken or adapted from the Mitiq paper. https://arxiv.org/pdf/2009.04417.pdf. This is related to #908 as it would give more immediate information on the techniques in Mitiq.

### PR DESCRIPTION
**Pull request by nathanshammah**
Close #944.
- Add workflow figures and their descriptions for ZNE, PEC and CDR. 
- Added section on CDR to be able to include the figure

By the way, I did not manage to add captions. The `:caption:` directive does not work.

### CODE REVIEW

File: docs/source/guide/guide-cdr.rst [hunk 1]
```diff
@@ -0,0 +1,27 @@
+.. cdr:
+
+************************
+Clifford Data Regression
+************************
+
+Clifford Data Regression (CDR) is a learning-based quantum error mitigation technique. It is based on the
```
Comment (rmlarose) [inline docs/source/guide/guide-cdr.rst:7]: ?

**Review by rmlarose — COMMENTED** [review]
Which guide (1-5) does `guide-cdr.rst` fit into w.r.t. the docs reorganization RFC in https://github.com/unitaryfund/mitiq/issues/908#issuecomment-926790261? 

IMO the images should go in 1. ("How do I use this technique?"), but `guide-cdr.rst` doesn't explain how to use CDR.

**Comment by nathanshammah** [comment]
> Which guide (1-5) does `guide-cdr.rst` fit into w.r.t. the docs reorganization RFC in [#908 (comment)](https://github.com/unitaryfund/mitiq/issues/908#issuecomment-926790261)?
> 
> IMO the images should go in 1. ("How do I use this technique?"), but `guide-cdr.rst` doesn't explain how to use CDR.

@rmlarose this PR is only involved with adding the workflow and technique figures. Especially for CDR, this is a minimal addition given that @L-P-B is working on the full theory. But to be able to add the figure I had to create the relative section.  The order of 1-5 is not relevant here.

**Comment by rmlarose** [comment]
> But to be able to add the figure I had to create the relative section. The order of 1-5 is not relevant here.

I don't see how the documentation reorganization we discussed for two weeks is not relevant. Maybe restructure one of the technique docs according to the RFC and add one of the figures there - probably in section 4: "What happens when I use this technique?".

**Comment by nathanshammah** [comment]
Thanks @andreamari. It took me a bit to work out how to link to a myst file from rst...

**Comment by andreamari** [comment]
Thanks @nathanshammah, I fixed a link in the last commit and all the rest is good for me.

@rmlarose: it would be good to have this in the 0.11 stable release. Too late?

**Comment by nathanshammah** [comment]
@andreamari I worked out the relative link fix, using a label and reference that works between rst and myst. It should be fine. I now just need to do the diff with your latest commit.

**Comment by nathanshammah** [comment]
The test failing on my latest commit is not due to the changes in this PR but a random value in the ZNE code.

**Comment by nathanshammah** [comment]
I rerun the CI and now tests pass.

**Pull request merged**